litmus-portal/graphql-server/pkg/k8s/cluster.go

```diff
@@ -0,0 +1,125 @@
+package k8s
+
+import (
+	"context"
+	"log"
+	"os"
+	"strconv"
+
+	appsv1 "k8s.io/api/apps/v1"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func CreateDeployment(litmusPortalNS, token string) (*appsv1.Deployment, error) {
+	deployerImage := os.Getenv("DEPLOYER_IMAGE")
+
+	clientset, err := GetGenericK8sClient()
+	if err != nil {
+		return nil, err
+	}
+
+	portalEndpoint, err := GetPortalEndpoint()
+	if err != nil {
+		return nil, err
+	}
+
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "self-deployer",
+			Labels:    map[string]string{"component": "self-deployer"},
+			Namespace: litmusPortalNS,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: func(i int32) *int32 { return &i }(1),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"component": "self-deployer",
+				},
+			},
+			Template: apiv1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"component": "self-deployer",
+					},
+				},
+				Spec: apiv1.PodSpec{
+					Containers: []apiv1.Container{
+						{
+							Name:            "deployer",
+							Image:           deployerImage,
+							ImagePullPolicy: "Always",
+							Env: []apiv1.EnvVar{
+								{
+									Name:  "SERVER",
+									Value: portalEndpoint,
+								},
+								{
+									Name:  "TOKEN",
+									Value: token,
+								},
+							},
+						},
+					},
+					ServiceAccountName: "litmus-svc-account",
+				},
+			},
+		},
+	}
+
+	// Create Deployment
+	log.Println("Creating deployment...")
+	result, err := clientset.AppsV1().Deployments(litmusPortalNS).Create(context.TODO(), deployment, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+func GetPortalEndpoint() (string, error){
+	var (
+		Nodeport int32
+		ExternalIP string
+		InternalIP string
+		LitmusPortalNS = os.Getenv("LITMUS_PORTAL_NAMESPACE")
+	)
+
+	clientset, err := GetGenericK8sClient()
+	if err != nil {
+		return "", err
+	}
+
+	podList, _ := clientset.CoreV1().Pods(LitmusPortalNS).List(context.TODO(), metav1.ListOptions{
+		LabelSelector: "component=litmusportal-server",
+	});
+
+	svc, err := clientset.CoreV1().Services(LitmusPortalNS).Get(context.TODO(), "litmusportal-server-service", metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	for _, port := range svc.Spec.Ports {
+		if port.Name == "graphql-server" {
+			Nodeport = port.NodePort
+		}
+	}
+
+	nodeIP, err := clientset.CoreV1().Nodes().Get(context.TODO(), podList.Items[0].Spec.NodeName, metav1.GetOptions{})
```

Comment:
This will not work in the ns scoped installation for porta?

Reply:
kubectl get node doesn't support namespace

Reply:
How will this work in ns mode then?

Reply:
Just like CRDs, it also requires admin permission to get nodes.

Reply:
For namespace mode- We can add a new variable (ClusterScope: Namespace/Cluster). If the user chooses `Namespace`, then he has to set an Endpoint ENV manually.  
Namespace scope implementation can be done later because we haven't support NS scope yet.